skipper.go

```diff
@@ -310,6 +309,18 @@ type Options struct {
 	// Custom data clients to be used together with the default etcd and Innkeeper.
 	CustomDataClients []routing.DataClient
 
+	// CustomHttpHandlerWrap provides ability to wrap http.Handler created by skipper.
+	// http.Handler is used for accepting incoming http requests.
+	// It allows to add additional logic (for example tracing) by providing a wrapper function
+	// which accepts original skipper handler as an argument and returns a wrapped handler
+	CustomHttpHandlerWrap func(http.Handler) http.Handler
+
+	// CustomHttpRoundTripperWrap provides ability to wrap http.RoundTripper created by skipper.
+	// http.RoundTripper is used for making outgoing requests (backends)
+	// It allows to add additional logic (for example tracing) by providing a wrapper function
+	// which accepts original skipper http.RoundTripper as an argument and returns a wrapped roundtripper
+	CustomHttpRoundTripperWrap func(http.RoundTripper) http.RoundTripper
```

Comment:
Maybe it's then better to call it `WithRoundTripper`?
@aryszka do you have a name suggestion?

Reply:
Or simply WrapTransport? I don't have a strong opinion, though.